airbyte/_processors/sql/snowflakecortex.py

```diff
@@ -0,0 +1,258 @@
+        if not self._catalog:
+            raise exc.AirbyteLibInternalError(
+                message="Catalog should exist but does not.",
+            )
+
+        matching_streams: list[ConfiguredAirbyteStream] = [
+            stream for stream in self._catalog.streams if stream.stream.name == stream_name
+        ]
+        if not matching_streams:
+            raise exc.AirbyteStreamNotFoundError(
+                stream_name=stream_name,
+                context={
+                    "available_streams": [stream.stream.name for stream in self._catalog.streams],
+                },
+            )
+
+        if len(matching_streams) > 1:
+            raise exc.AirbyteLibInternalError(
+                message="Multiple streams found with same name.",
+                context={
+                    "stream_name": stream_name,
+                },
+            )
+
+        return matching_streams[0]
+
+    @overrides
+    def _ensure_compatible_table_schema(
+        self,
+        stream_name: str,
+        *,
+        raise_on_error: bool = True,
+    ) -> bool:
+        # to-do: SQLAlchemy is not compatible with vector type. Implemenbt using Snowflake python connector
+        return True
+
+    def _finalize_state_messages(
+        self,
+        stream_name: str,
+        state_messages: list[AirbyteStateMessage],
+    ) -> None:
+        # to-do: need to override since we are not using catalog manager. Do we want to write state messages in this scenario ?
+        pass
+
+    def _get_json_schema_from_catalog(
+        self,
+        stream_name: str,
+    ) -> dict[str, Any]:
+        return self._get_stream_config(stream_name).stream.json_schema
+
+    @overrides
+    def _write_files_to_new_table(
+        self,
+        files: list[Path],
+        stream_name: str,
+        batch_id: str,
+    ) -> str:
+        """Write files to a new table."""
+        temp_table_name = self._create_table_for_loading(
+            stream_name=stream_name,
+            batch_id=batch_id,
+        )
+        internal_sf_stage_name = f"@%{temp_table_name}"
+
+        def path_str(path: Path) -> str:
+            return str(path.absolute()).replace("\\", "\\\\")
+
+        put_files_statements = "\n".join(
+            [f"PUT 'file://{path_str(file_path)}' {internal_sf_stage_name};" for file_path in files]
+        )
+        print("put file statements")
+        print(put_files_statements)
+        self._execute_sql(put_files_statements)
+        columns_list = [
+            self._quote_identifier(c)
+            for c in list(self._get_sql_column_definitions(stream_name).keys())
+        ]
+        files_list = ", ".join([f"'{f.name}'" for f in files])
+        columns_list_str: str = indent("\n, ".join(columns_list), " " * 12)
+
+        # to-do: move following into a separate method, then don't need to override the whole method.
```

Comment:
We need to read embedding data suffixed by ::Vector(float, xx) for it to be written like that to Snowflake. This is the only change in this method.